File: app/src/main/java/ru/mamykin/foreignbooksreader/preferences/PreferencesManager.java
package ru.mamykin.foreignbooksreader.preferences;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

import ru.mamykin.foreignbooksreader.ReaderApp;
import ru.mamykin.foreignbooksreader.preferences.PreferenceNames;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
public class PreferencesManager implements PreferenceNames {
    @Inject
    protected Context context;

    public PreferencesManager() {
        ReaderApp.getComponent().inject(this);
    }

    private SharedPreferences getSharedPreferences() {
        return context.getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE);
    }

    public void putBoolean(String key, boolean value) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value).apply();
    }

    public boolean getBoolean(String key, boolean defValue) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getBoolean(key, defValue);
    }

    public boolean getBoolean(String key) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getBoolean(key, false);
    }

    public void putString(String key, String value) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value).apply();
    }

    public String getString(String key, String defValue) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getString(key, defValue);
    }

    public String getString(String key) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getString(key, null);
    }

    public void putInt(String key, int value) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt(key, value).apply();
    }

    public int getInt(String key, int defValue) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getInt(key, defValue);
    }

    public int getInt(String key) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getInt(key, -1);
    }

    public void putFloat(String key, float value) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putFloat(key, value).apply();
    }

    public float getFloat(String key, float defValue) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getFloat(key, defValue);
    }

    public float getFloat(String key) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.getFloat(key, -1f);
    }

    public void removeValue(String key) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.remove(key).apply();
    }

    public boolean contains(String prefName) {
        SharedPreferences sharedPreferences = getSharedPreferences();
        return sharedPreferences.contains(prefName);
    }
}

File: app/src/main/java/ru/mamykin/foboreader/data/storage/PreferencesManager.kt
package ru.mamykin.foboreader.data.storage

import android.content.Context
import android.content.SharedPreferences
import ru.mamykin.foboreader.ReaderApp
import javax.inject.Inject

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
class PreferencesManager : PreferenceNames {
    @Inject
    lateinit var context: Context

    private val sharedPreferences: SharedPreferences
        get() = context!!.getSharedPreferences(PreferenceNames.PREFERENCES_FILE, Context.MODE_PRIVATE)

    init {
        ReaderApp.component.inject(this)
    }

    fun putBoolean(key: String, value: Boolean) {
        val sharedPreferences = sharedPreferences
        val editor = sharedPreferences.edit()
        editor.putBoolean(key, value).apply()
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getBoolean(key, defValue)
    }

    fun getBoolean(key: String): Boolean {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getBoolean(key, false)
    }

    fun putString(key: String, value: String) {
        val sharedPreferences = sharedPreferences
        val editor = sharedPreferences.edit()
        editor.putString(key, value).apply()
    }

    fun getString(key: String, defValue: String): String? {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getString(key, defValue)
    }

    fun getString(key: String): String? {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getString(key, null)
    }

    fun putInt(key: String, value: Int) {
        val sharedPreferences = sharedPreferences
        val editor = sharedPreferences.edit()
        editor.putInt(key, value).apply()
    }

    fun getInt(key: String, defValue: Int): Int {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getInt(key, defValue)
    }

    fun getInt(key: String): Int {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getInt(key, -1)
    }

    fun putFloat(key: String, value: Float) {
        val sharedPreferences = sharedPreferences
        val editor = sharedPreferences.edit()
        editor.putFloat(key, value).apply()
    }

    fun getFloat(key: String, defValue: Float): Float {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getFloat(key, defValue)
    }

    fun getFloat(key: String): Float {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.getFloat(key, -1f)
    }

    fun removeValue(key: String) {
        val sharedPreferences = sharedPreferences
        val editor = sharedPreferences.edit()
        editor.remove(key).apply()
    }

    operator fun contains(prefName: String): Boolean {
        val sharedPreferences = sharedPreferences
        return sharedPreferences.contains(prefName)
    }
}